core_types/src/json.rs

```diff
@@ -57,17 +57,16 @@ impl<T: Serialize> From<Vec<T>> for JsonString {
 
 impl<T: Into<JsonString>, E: Into<JsonString>> From<Result<T, E>> for JsonString {
     fn from(result: Result<T, E>) -> JsonString {
-        JsonString::from(
-            serde_json::to_string(&match result {
+        JsonString::from(match result {
```

Comment:
You could replace this encapsulating call to `JsonString::from` with a postfix `.into()`. Because every `From` implementation automatically generates an `Into` implementation